app/models/tag.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class Tag < ApplicationRecord
-  has_many  :article_tags
-  has_many  :articles, through: :article_tags
+  has_many  :article_tags, dependent: :nullify
+  has_many  :articles, through: :article_tags, dependent: :nullify
```

Comment:
i think it's not logical

so when tag is deleted then article_tag stays in db
but why we need that article_tag??

Reply:
I agree with you. 
Do I need to change `has_many :article_tags, dependent: :nullify` to `has_many :article_tags, dependent: :destroy`?
But leave the 5th line unchanged?

Reply:
Dependents fixed.